### ISSUE TITLE
Tabs panel combination problems (v2)

### ISSUE THREAD

**Issue by inventionate** [eminiarts/nova-tabs#188]
First of all, thank you for the great package. I am very happy that there is now a Nova 4 version. Unfortunately, it seems that it is no longer possible to combine a tab panel with other fields and panels? Following setup results in a cluttered overall panel that is not usable:

```php
use HasTabs;

public function fields(NovaRequest $request)
{
        return [
            Text::make('Name', 'name')
                ->sortable()
                ->onlyOnIndex(),

            Text::make('Name')
                ->rules('required', 'string')
                ->creationRules('unique:schools,name')
                ->updateRules('unique:schools,name,{{resourceId}}')
                ->sortable()
                ->hideFromIndex(),

            Panel::make('Team', $this->teamFields()),

            Tabs::make('Portrait', [
                'Community'     => $this->CommunityFields(),
                'Context'       => $this->contextFields(),
                'Profile'       => $this->profileFields(),
                'Network'.      => $this->networkFields(),
            ])
                ->showTitle(),

            BelongsToMany::make('Projects', 'projects', 'App\Nova\Project'),
       ]
}

```

Under version 1 everything worked without problems. Maybe I am doing something wrong?

**Comment by marcfilipan** [comment]
Does the panel show without the fields or simply doesn't show at all?

**Comment by inventionate** [comment]
All fields are shown, but they are messed up. Fields which are not in a panel or relations (belongstomany etc.) are all placed in a new panel. Under the new panel are the tabs and explicit panels (other example code but same result):

![Bildschirmfoto 2022-04-30 an 23 26 43](https://user-images.githubusercontent.com/872787/166123221-b20a6d38-9c87-45e1-a599-65b0d5a83a87.png)

**Comment by marcfilipan** [comment]
I'll look into it tomorrow, think i might be overwriting the component for panels under tabs.

That the fields who aren't in tabs or panels at all show up in a new panel, is simply how Nova works.

**Comment by inventionate** [comment]
Thank you! And yes, panels aren't the problem, but putting all into one is. Here is a screenshot when I don't load the HasTabs trait. I think it is definitely about an overwrite (nearly as excepted, but missing tab panel title):

![Bildschirmfoto 2022-05-01 an 07 23 22](https://user-images.githubusercontent.com/872787/166133447-4a53983e-4035-400e-ba75-62ad8c691588.png)

**Comment by marcfilipan** [comment]
Please post matching code to the screenshots you share so I can recreate the situation. The only issue I can find at the moment is that you can't use panels inside a tab, but other then that everything seems to work fine.

**Comment by inventionate** [comment]
No problem. The traits are just arrays of fields and it's no difference if I use them or just array of fields directly:

```php
<?php

namespace App\Nova;

use App\Nova\SchoolPortraitFields\AllDayService;
use App\Nova\SchoolPortraitFields\Context;
use App\Nova\SchoolPortraitFields\Corona;
use App\Nova\SchoolPortraitFields\Equipment;
use App\Nova\SchoolPortraitFields\FeedbackTransfer;
use App\Nova\SchoolPortraitFields\FormatOfPromotion;
use App\Nova\SchoolPortraitFields\FurtherTraining;
use App\Nova\SchoolPortraitFields\MissionStatement;
use App\Nova\SchoolPortraitFields\Network;
use App\Nova\SchoolPortraitFields\ParticipationLemas;
use App\Nova\SchoolPortraitFields\Profile;
use App\Nova\SchoolPortraitFields\SchoolCommunity;
use Eminiarts\Tabs\Tabs;
use Eminiarts\Tabs\Traits\HasTabs;
use Laravel\Nova\Fields\BelongsToMany;
use Laravel\Nova\Fields\Boolean;
use Laravel\Nova\Fields\HasMany;
use Laravel\Nova\Fields\Line;
use Laravel\Nova\Fields\Markdown;
use Laravel\Nova\Fields\Number;
use Laravel\Nova\Fields\Place;
use Laravel\Nova\Fields\Select;
use Laravel\Nova\Fields\Stack;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Http\Requests\NovaRequest;
use Laravel\Nova\Panel;

class School extends Resource
{
    use SchoolCommunity,
        Context,
        MissionStatement,
        AllDayService,
        Equipment,
        Profile,
        FurtherTraining,
        Network,
        ParticipationLemas,
        FormatOfPromotion,
        FeedbackTransfer,
        Corona,
        HasTabs;

    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     *
     * @return array
     */
    public function fields(NovaRequest $request)
    {
        return [
            Text::make('Name')
                ->rules('required', 'string')
                ->creationRules('unique:schools,name')
                ->updateRules('unique:schools,name,{{resourceId}}')
                ->sortable()
                ->hideFromIndex(),

            Select::make('Schulart', 'type')
                ->rules('required', 'string')
                ->options([
                    'Primarstufe'           => 'Primarstufe',
                    'Weiterführende Schule' => 'Weiterführende Schule',
                ])
                ->sortable(),

            Panel::make('Schulleiter/in', $this->headmasterFields())
                ->collapsible(),

            Tabs::make('Schulportrait', [
                'Schulgemeinschaft'              => $this->schoolCommunityFields(),
                'Lage'                           => $this->contextFields(),
                'Ganztagsbetrieb'                => $this->allDayServiceFields(),
                'Ausstattung'                    => $this->equipmentFields(),
                'Profil'                         => $this->profileFields(),
                'Leitbild'                       => $this->missionStatementFields(),
                'Fort/Weiterbildung'             => $this->furtherTrainingFields(),
                'Netzwerk'                       => $this->networkFields(),
                'Teilnahme an LemaS/Ziele'       => $this->participationLemasFields(),
                'Formate der Begabungsförderung' => $this->formatOfPromotionFields(),
                'Transferphase'                  => $this->feedbackTransferFields(),
                'Corona'                         => $this->coronaFields(),
            ])
                ->showTitle(),

            BelongsToMany::make('Teilprojekte', 'projects', 'App\Nova\Project')
                ->singularLabel('Teilprojekt')
                ->fields(function () {
                    return [
                        Boolean::make('Zusätzlich begleitet', 'accompanied'),
                        Boolean::make('Adaptiv begleitet', 'adaptive'),
                    ];
                })
                ->searchable(),

            BelongsToMany::make('Wissenschaftler/-innen', 'members', 'App\Nova\Member')
                ->searchable()
                ->singularLabel('Wissenschaftler/-in'),

            Text::make('Sekreteriat E-Mail', 'office_email')
                ->sortable()
                ->nullable()
                ->rules('email', 'max:254', 'nullable')
                ->hideFromIndex(),

            Text::make('Telefonnummer', 'phone')
                ->rules('digits_between:4,20', 'nullable')
                ->nullable()
                ->hideFromIndex(),

            Text::make('Website')
                ->nullable()
                ->hideFromIndex()
                ->rules('url', 'nullable'),

            Place::make('Adresse', 'address')->hideFromIndex(),

            Text::make('Stadt', 'city')->hideFromIndex(),

            Text::make('Bundesland', 'state')->sortable(),

            Text::make('Postleitzahl', 'postal_code')->hideFromIndex(),

            Number::make('long')
                ->step(0.000000000000001)
                ->canSee(function ($request) {
                    return $request->user()->type === 'admin';
                })
                ->hideFromIndex(),

            Number::make('lat')
                ->step(0.000000000000001)
                ->canSee(function ($request) {
                    return $request->user()->type === 'admin';
                })
                ->hideFromIndex(),

            Text::make('Matchingergebnis KM1', 'matching_original_km1')
                ->hideWhenCreating()
                ->hideWhenUpdating()
                ->hideFromIndex(),

            Text::make('Matchingergebnis KM2', 'matching_original_km2')
                ->hideWhenCreating()
                ->hideWhenUpdating()
                ->hideFromIndex(),

            Markdown::make('Kommentar', 'comment')
                ->nullable()
                ->alwaysShow()
                ->hideFromIndex(),

            HasMany::make('Schulteam', 'schoolContacts', 'App\Nova\SchoolContact')
                ->singularLabel('Schulteam'),

            BelongsToMany::make('Produkte', 'products', 'App\Nova\Product')
                ->singularLabel('Produkt')
                ->searchable(),
        ];
    }

}

```

Thank you!

**Comment by marcfilipan** [comment]
https://github.com/eminiarts/nova-tabs/releases/tag/2.0.1

Give it a try please @inventionate and let me know if this gives you the expected result

**Comment by inventionate** [comment]
Works great! Thank you! Just one small thing. It seems that the tabs panel misses some top margin between heading and panel compared to normal nova panels. If I add mt-3 py-2 to div.tab-card it's perfect.

**Comment by marcfilipan** [comment]
The padding issue was only when not showing it as toolbar, fixed now in 2.0.2, thanks for finding it!

**Comment by inventionate** [comment]
Great! Thanks for the fast fixes and the wonderful package!
